File: src/main/kotlin/wowsft/model/gameparams/ship/BattleLevels.java
package wowsft.model.gameparams.ship;

import wowsft.config.WoWSFT;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;

@WoWSFT
@JsonIgnoreProperties(ignoreUnknown = true)
public class BattleLevels
{
    @JsonProperty("COOPERATIVE")
    private List<Integer> cooperative;
    @JsonProperty("PVP")
    private List<Integer> pvp;
}

File: src/main/kotlin/wowsft/model/gameparams/ship/BattleLevels.kt
package wowsft.model.gameparams.ship

import wowsft.config.WoWSFT
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonAlias

@WoWSFT
@JsonIgnoreProperties(ignoreUnknown = true)
class BattleLevels {
    @JsonAlias("COOPERATIVE")
    val cooperative = ArrayList<Int>()
    @JsonAlias("PVP")
    val pvp = ArrayList<Int>()
}
